### PR TITLE
Partially apply `#[deny(unsafe_op_in_unsafe_fn)]`

### PR DESCRIPTION
**Pull request by workingjubilee**
This doesn't cover the entire crate due to the issue noted in https://github.com/tcdi/pgx/pull/707.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -73,22 +80,26 @@ CREATE FUNCTION plrust_call_handler() RETURNS language_handler
...
     ) -> eyre::Result<pg_sys::Datum> {
-        let fn_oid = fcinfo
-            .as_ref()
-            .ok_or(PlRustError::NullFunctionCallInfo)?
-            .flinfo
-            .as_ref()
-            .ok_or(PlRustError::NullFmgrInfo)?
-            .fn_oid;
-        let retval = plrust::evaluate_function(fn_oid, fcinfo)?;
+        let fn_oid = unsafe {
```
Comment (BradyBonnette) [inline src/lib.rs:88]: Does this need  `// SAFETY` documentation?

File: src/lib.rs [hunk 2]
```diff
@@ -73,22 +80,26 @@ CREATE FUNCTION plrust_call_handler() RETURNS language_handler
...
+            fcinfo
+                .as_ref()
+                .ok_or(PlRustError::NullFunctionCallInfo)?
+                .flinfo
+                .as_ref()
+        }
+        .ok_or(PlRustError::NullFmgrInfo)?
+        .fn_oid;
+        let retval = unsafe { plrust::evaluate_function(fn_oid, fcinfo)? };
```
Comment (BradyBonnette) [inline src/lib.rs:97]: Does this need  `// SAFETY` documentation?

File: src/lib.rs [hunk 3]
```diff
@@ -97,27 +108,33 @@ unsafe fn plrust_call_handler(fcinfo: pg_sys::FunctionCallInfo) -> pg_sys::Datum
...
+            !pg_sys::CheckFunctionValidatorAccess(
+                flinfo.fn_oid,
+                pg_getarg(fcinfo.as_ptr(), 0).unwrap(),
+            )
+        } {
             return Err(PlRustError::CheckFunctionValidatorAccess)?;
         }
 
-        plrust::unload_function(fn_oid);
+        unsafe { plrust::unload_function(fn_oid) };
```
Comment (BradyBonnette) [inline src/lib.rs:130]: Does this need `// SAFETY` documentation?

File: src/lib.rs [hunk 4]
```diff
@@ -132,7 +149,7 @@ unsafe fn plrust_validator(fn_oid: pg_sys::Oid, fcinfo: pg_sys::FunctionCallInfo
         Ok(())
     }
 
-    match plrust_validator_inner(fn_oid, fcinfo) {
+    match unsafe { plrust_validator_inner(fn_oid, fcinfo) } {
```
Comment (BradyBonnette) [inline src/lib.rs:152]: Does this need `// SAFETY` documentation?

**Review by BradyBonnette — COMMENTED** [review]
@workingjubilee Apologies for the bombardment of `// SAFETY` documentation questions. If none of them apply then you can resolve the conversations. I just wanted to be sure you weren'r forgetting unsafe doc in a few spots.

**Comment by workingjubilee** [comment]
So, in all of these cases: yes, but I don't feel fully confident to give the safety qualifications fully as I didn't write the code initially in these cases, so the introduction of `unsafe { }` changes nothing material about the code but giving false/bad documentation would cause them to pass lints they should not.

**Comment by BradyBonnette** [comment]
Fair!

**Review by BradyBonnette — APPROVED** [review]

**Pull request merged**